ksql-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -175,69 +246,64 @@ public LogicalSchema withAlias(final String alias) {
    * @return the schema without any aliases in the field name.
    */
   public LogicalSchema withoutAlias() {
-    final SchemaBuilder newSchema = SchemaBuilder
-        .struct()
-        .name(schema.name());
-
-    for (final Field field : schema.fields()) {
-      final String unaliased = SchemaUtil.getFieldNameWithNoAlias(field.name());
-      newSchema.field(unaliased, field.schema());
+    if (!alias.isPresent()) {
+      throw new IllegalStateException("Not aliased");
     }
 
-    return LogicalSchema.of(newSchema.build());
+    return new LogicalSchema(
+        removeAlias(implicitSchema),
+        removeAlias(keySchema),
+        removeAlias(valueSchema),
+        Optional.empty()
+    );
   }
 
   /**
-   * Add implicit fields to the schema.
+   * Copies implicit and key fields to the value schema.
    *
-   * <p>Implicit fields are:
-   * <ol>
-   * <li>{@link SchemaUtil#ROWTIME_NAME}</li>
-   * <li>{@link SchemaUtil#ROWKEY_NAME}</li>
-   * </ol>
+   * <p>If the fields already exist in the value schema the function returns the same schema.
    *
-   * <p>If the implicit fields already exist, the function returns the same schema.
-   *
-   * <p><b>NOTE:</b> the function does NOT take any aliases in the fields into account
-   *
-   * @return the new schema with the (unaliased) implicit fields added.
+   * @return the new schema.
    */
-  public LogicalSchema withImplicitFields() {
+  public LogicalSchema withImplicitAndKeyFieldsInValue() {
```

Comment:
Foreshadowing: Do we have plans to remove these methods completely? I feel like LogicalSchema should always have it with implicit key fields in value - the only reason you wouldn't want it is for serialization, which should be the concern of some other code.

Reply:
I disagree. I think in time the value should never have the key and implicit fields in the value schema.  They are key and implicit fields, not value fields.   I think this will all come out in the wash when we look to improve query semantics.